appserver/admin/admin-core/pom.xml

```diff
@@ -65,22 +65,24 @@
             <artifactId>common-util</artifactId>
             <version>${project.version}</version>
         </dependency>
-
-        <!-- test dependencies -->
         <dependency>
             <groupId>fish.payara.server.internal.test-utils</groupId>
             <artifactId>utils</artifactId>
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
-            <dependency>
-                <groupId>jakarta.el</groupId>
-                <artifactId>jakarta.el-api</artifactId>
-            </dependency>
-            <dependency>
-                <groupId>org.glassfish</groupId>
-                <artifactId>jakarta.el</artifactId>
-            </dependency>
+        <dependency>
+            <groupId>jakarta.el</groupId>
+            <artifactId>jakarta.el-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish</groupId>
+            <artifactId>jakarta.el</artifactId>
+        </dependency>
+        <dependency>
```

Comment:
@lprimak doesn't like it, we had a long discussion yesterday evening. We didn't find acceptable conclusion for both, so this morning I offered him, that I will summarize it here and people will decide which variant is better.

Lenny says it is a duplicity and it is better to have junit4 as a default mandatory dependency in common parent. 

That goes strictly against my experiences with especially large projects, because:

- it is a best practice to **declare in artifact pom file all dependencies directly used** despite they are brought as a transitive dependency too. 
- [maven dependency plugin](https://maven.apache.org/plugins/maven-dependency-plugin/plugin-info.html) has a goal to report used but undeclared transitive dependencies, some other projects use it in verify phase to keep sane dependency management.
- the shared dependency brought the JUnit4 dependency even to maven modules which doesn't need it, because they use JUnit5, TestNG or they simply don't contain any tests.
- **usage**: I open pom, I see dependencies, I know them ... or I don't and I have to use some smarter tool.
- both JUnit4 and JUnit5 contain @Test annotation - the most common mistake is using the old one and developer even doesn't notice that his test wasn't executed (most editors add imports automatically).
- four lines are not reported as a duplicity by any tool available (cpd plugin uses 100 tokens as a default), despite in git diff it looks ugly,
- ie. common-util dependency is in 90 pom.xml files and it is alright, probably
- btw license header is much longer and it is in each and every file and we have to live with that ...

So, these are all arguments I have to defend this. The change simply implicates from starting with JUnit5 and moving away from JUnit4 in dependant pull requests one by one module.

**So I am asking community to decide, which variant is better:**
- thumb up to accept it as it is
- thumb down to put back the shared mandatory JUnit4 dependency where it was and remove all those junit blocks.

I will respect the result.

Reply:
Can not find related commits, but we have chosen jupiter-vintage way for old code that we want to leave as is and pure junit-jupiter for new tests.

Reply:
Yes, that is a temporary [alternative](https://junit.org/junit5/docs/current/user-guide/#migrating-from-junit4) for JUnit4, but the discussion here is about managing complexity.

Reply:
@dmatej you mis-characterize this as a transitive dependency. It is not. It is a parent POM dependency. There is a difference. Parent POM dependency is not an anti-pattern.

Reply:
My issue that there are 400+ file changes in this PR just to add JUnit dependency to every single POM that has unit tests. This is a massive code duplication. Also, this does not belong in this PR. If anything, this belongs in a separate PR, and then this can be voted on.

You are trying to mash different, unrelated changes into a single PR and then trying to make people "take it or leave it" all-or-nothing decision.
This is unfair, and goes against all sort of standard, best practices